sdk/identity/Azure.Identity/src/EnvironmentVariables.cs

```diff
@@ -10,32 +10,42 @@ namespace Azure.Identity
 {
     internal class EnvironmentVariables
     {
-        public static string Username => Environment.GetEnvironmentVariable("AZURE_USERNAME");
-        public static string Password => Environment.GetEnvironmentVariable("AZURE_PASSWORD");
-        public static string TenantId => Environment.GetEnvironmentVariable("AZURE_TENANT_ID");
+        public static string Username => GetNonEmptyStringOrNull(Environment.GetEnvironmentVariable("AZURE_USERNAME"));
+        public static string Password => GetNonEmptyStringOrNull(Environment.GetEnvironmentVariable("AZURE_PASSWORD"));
+        public static string TenantId => GetNonEmptyStringOrNull(Environment.GetEnvironmentVariable("AZURE_TENANT_ID"));
         public static List<string> AdditionallyAllowedTenants => (Environment.GetEnvironmentVariable("AZURE_ADDITIONALLY_ALLOWED_TENANTS") ?? string.Empty).Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries).ToList();
```

Comment:
Should this also use `GetNonEmptyStringOrNull`

Reply:
In this case I actually want the value to be empty string rather than null. As you can see if the value of the environment variable is null actually switch it to string .Empty:

```c#
(Environment.GetEnvironmentVariable("AZURE_ADDITIONALLY_ALLOWED_TENANTS") ?? string. Empty)
```
I'm doing this so split will return an empty list if the value of the env var is null or string. Empty.